.gitignore

```diff
@@ -34,3 +34,4 @@ temp.mlir
 tags
 test-semantics/*.mlir
 test-semantics/*.llvm
+*.olean
```

Comment:
This should be a separate PR. Happy for it to be merged without review.

Reply:
Pushed to main :)